plasmapy/atomic/particle_class.py

```diff
@@ -1015,6 +1015,34 @@ def binding_energy(self) -> u.Quantity:
 
         return nuclear_binding_energy.to(u.J)
 
+    @property
+    def mass_energy(self) -> u.Quantity:
+        """
+        Return the nuclear mass energy in joules.
+
+        This attribute will raise an
+        `~plasmapy.utils.InvalidIsotopeError` if the particle is not a
+        nucleon or isotope.
+
+        """
+        if self._attributes['baryon number'] == 1:
+            return 0 * u.J
+
+        if not self.isotope:
+            raise InvalidIsotopeError(
+                f"The nuclear binding energy may only be calculated for nucleons and isotopes.")
+
+        number_of_protons = self.atomic_number
+        number_of_neutrons = self.mass_number - self.atomic_number
+
+        mass_of_protons = number_of_protons * const.m_p
+        mass_of_neutrons = number_of_neutrons * const.m_n
+
+        mass_of_nucleons = mass_of_protons + mass_of_neutrons
+        nuclear_mass_energy = mass_of_nucleons * const.c ** 2
+
```

Comment:
I'm pretty sure this whole calculation could be simplified using https://github.com/Griffintaur/PlasmaPy/blob/998d5b5b76ceebcac42f3c3ad2d0aacaa6a7e8e6/plasmapy/atomic/nuclear.py#L261-L272 this snippet, and it'd let you skip the exception, which I don't think is necessary here.